src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -0,0 +1,488 @@
+                throw new MsalClientException(MsalError.CannotInvokeBroker, msg);
+            }
+        }
+
+        private string GetRedirectUriForBroker()
+        {
+            string packageName = Application.Context.PackageName;
+
+            // First available signature. Applications can be signed with multiple
+            // signatures.
+            string signatureDigest = GetCurrentSignatureForPackage(packageName);
+            if (!string.IsNullOrEmpty(signatureDigest))
+            {
+                return string.Format(CultureInfo.InvariantCulture, "{0}://{1}/{2}", RedirectUriScheme,
+                    packageName.ToLowerInvariant(), signatureDigest);
+            }
+
+            return string.Empty;
+        }
+
+        private string GetCurrentSignatureForPackage(string packageName)
+        {
+            try
+            {
+                PackageInfo info = Application.Context.PackageManager.GetPackageInfo(packageName,
+                    PackageInfoFlags.Signatures);
+                if (info != null && info.Signatures != null && info.Signatures.Count > 0)
+                {
+                    Signature signature = info.Signatures[0];
+                    MessageDigest md = MessageDigest.GetInstance("SHA");
+                    md.Update(signature.ToByteArray());
+                    return Convert.ToBase64String(md.Digest(), Base64FormattingOptions.None);
+                    // Server side needs to register all other tags. ADAL will
+                    // send one of them.
+                }
+            }
+            catch (PackageManager.NameNotFoundException)
+            {
+                _logger.Info("Calling App's package does not exist in PackageManager");
+            }
+            catch (NoSuchAlgorithmException)
+            {
+                _logger.Info("Digest SHA algorithm does not exists");
+            }
+
+            return null;
+        }
+
+        private Intent GetInteractiveBrokerIntent(IDictionary<string, string> brokerPayload, Intent brokerIntent)
+        {
+            ValidateBrokerRedirectURI(brokerPayload);
+            BrokerRequest request = new BrokerRequest
+            {
+                Authority = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Authority),
+                Scopes = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Scope),
+                RedirectUri = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.RedirectUri),
+                ClientId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.ClientId),
+                ClientAppName = Application.Context.PackageName,
+                ClientAppVersion = Application.Context.PackageManager.GetPackageInfo(Application.Context.PackageName, PackageInfoFlags.MatchAll).VersionName,
+                ClientVersion = "4.4.0",
+                CorrelationId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.CorrelationId),
+                Prompt = "NONE"
```

Comment:
>"NONE" [](start = 25, length = 6)

const?

Reply:
No, this will be pulled from the interactive parameters